### ISSUE TITLE
MISPWarninglists analyzer doesn't work [Bug]

### ISSUE THREAD

**Issue by NailBash** [TheHive-Project/Cortex-Analyzers#827]
**Describe the bug**
MISPWarninglists analyzer doesn't work

**To Reproduce**
1. Download MISP warninglists
2. Configure analyzer
3. See error message "Traceback (most recent call last): File "/opt/Cortex-Analyzers/analyzers/MISPWarningLists/mispwarninglists.py", line 202, in <module> MISPWarninglistsAnalyzer().run() File "/opt/Cortex-Analyzers/analyzers/MISPWarningLists/mispwarninglists.py", line 91, in run for list in self.warninglists:TypeError: 'NoneType' object is not iterable"

![изображение](https://user-images.githubusercontent.com/29750691/88768915-2dfb5600-d195-11ea-97a9-066812670fc5.png)

**Expected behavior**
Normal work of analyzer

**Work environment**
Cortex runs on Centos 7.

**Comment by dadokkio** [comment]
Hello, from the error I can suppose that you are not using a db but local files. It's that correct?
Using locals file the analyzer require that you have cloned locally the misp-warninglist repo in the path specified in the settings.
If you have done this step double check if the path has been set properly.

**Comment by NailBash** [comment]
> 
> 
> Hello, from the error I can suppose that you are not using a db but local files. It's that correct?
> Using locals file the analyzer require that you have cloned locally the misp-warninglist repo in the path specified in the settings.
> If you have done this step double check if the path has been set properly.

**Comment by NailBash** [comment]
> 
> 
> Hello, from the error I can suppose that you are not using a db but local files. It's that correct?
> Using locals file the analyzer require that you have cloned locally the misp-warninglist repo in the path specified in the settings.
> If you have done this step double check if the path has been set properly.

Hello, yes I'm using local files, I'm cloned files into my local filesystem
This is my folder with warninglists
![изображение](https://user-images.githubusercontent.com/29750691/88783222-f34ee900-d1a7-11ea-95b8-7cfb69d89c04.png)
This is configuration in cortex
![изображение](https://user-images.githubusercontent.com/29750691/88783441-390bb180-d1a8-11ea-80a3-55a99b25c2bd.png)
I think, that path is correct, I checked it (Copied code which reads the warninglists and launched it by python3. It worked well and readed files)

**Comment by dadokkio** [comment]
The path seems to be correct, so we can start to check why the loop don't add anything to warninglists.
If you run this simple code on the machine:

```
from glob import glob
import io

files = glob("{}/lists/*/*.json".format("/home/cortex/MISP_warninglists/misp-warninglists"))
for file in files:
	with io.open(file, "r") as fh:
		print(file)
```
Does it print something?

**Comment by NailBash** [comment]
> 
> 
> The path seems to be correct, so we can start to check why the loop don't add anything to warninglists.
> If you run this simple code on the machine:
> 
> ```
> from glob import glob
> import io
> 
> files = glob("{}/lists/*/*.json".format("/home/cortex/MISP_warninglists/misp-warninglists"))
> for file in files:
> 	with io.open(file, "r") as fh:
> 		print(file)
> ```
> 
> Does it print something?

![изображение](https://user-images.githubusercontent.com/29750691/88789211-e504cb00-d1af-11ea-9147-108272af1a1b.png)

**Comment by dadokkio** [comment]
Ok, then probably is my code wrongly assuming you wants to use db.
Can you check if using this code everything works?

`self.warninglists = self.readwarninglists() if not conn or not USE_DB else None
`

**Comment by NailBash** [comment]
> 
> 
> Ok, then probably is my code wrongly assuming you wants to use db.
> Can you check if using this code everything works?
> 
> `self.warninglists = self.readwarninglists() if not conn or not USE_DB else None `

![изображение](https://user-images.githubusercontent.com/29750691/88797824-62830800-d1bd-11ea-90e4-02b2223e8446.png)

![изображение](https://user-images.githubusercontent.com/29750691/88797864-6e6eca00-d1bd-11ea-8199-690c6b20d474.png)

![изображение](https://user-images.githubusercontent.com/29750691/88797882-7595d800-d1bd-11ea-9703-5a76e13250d0.png)

**Comment by dadokkio** [comment]
Ok, can now test the change I proposed?

**Comment by NailBash** [comment]
> Ok, can now test the change I proposed?

![image](https://user-images.githubusercontent.com/29750691/88804747-cb6f7d80-d1c7-11ea-8eb7-aae7096f427a.png)

**Comment by NailBash** [comment]
![image](https://user-images.githubusercontent.com/29750691/88808442-55214a00-d1cc-11ea-8a60-c9b985dd5998.png)

It's ok for local filesystem?

**Comment by dadokkio** [comment]
Yes, this is the reason why we decide to add postgres as backend.
The first import in the db takes some times but then it returns answer in millisec.

We used to run the analyzer with  hundreds of observables and it ran for a lot of hours.

**Comment by garanews** [comment]
fixed in 2.9.0
